Makefile

```diff
@@ -420,7 +424,7 @@ $(LIBWASI_EMULATED_PROCESS_CLOCKS_OBJS): CFLAGS += \
     -I$(LIBC_BOTTOM_HALF_CLOUDLIBC_SRC)
 
 include_dirs:
-	$(RM) -r "$(SYSROOT)"
+	$(RM) -r "$(SYSROOT_INC)"
```

Comment:
I think its we are building the sysroot from scratch we probably want remove the entire thing somewhere. (doesn't have to be here).

Reply:
@sbc100 well the idea with this change is that there may not be a "sysroot" anymore --- these dirs are independent can couldn't have `/` even as their most recent common parent. I just didn't rename the variables to avoid churn.

Reply:
I think we should still default to the normal "build an entire sysroot" approach.   This new approach your are proposing probably isn't want to want for wasi-sdk.. or, I imagine, what most folks building wasi-libc will want, so maybe it should opt-in?

Reply:
BTW are you actually planning to use this approach of having the sysroot split up like this, or is it just to conform to the norms of your build system?   i.e. in practice, will your package of wasm-libc still have sysroot structure (i.e. will your users be running clang with --sysroot to access wasi-libc?  If not, how will they be running it?)

Reply:
The concept of "sysroot" is something clang understands so I think it will still be useful if in the final thing you package has a sysroot structure.. but maybe I'm misunderstanding something.

Reply:
Sorry I just saw these now. We do use this, and we do install in directories with a more distant common ancestor.

Reply:
@sbc100 you may be interested in https://github.com/mesonbuild/meson/pull/9903#issuecomment-1071105428 which is us trying to do this thing for Meson (and therefore fix it for a lot of packages in one swoop). A few other folks with similar needs chimed in there too.

Reply:
I wonder if you can get away with building the entire sysroot like we do (locally in a single root that we control), and then just modifying the `install` phase.    Technically its the install phase you care about right?  Its not until we get to installation that we want to honor the build systems target directories is it?

Reply:
If we allow SYSROOT_INC to be externally controlled it might not be something we can just remove with `rm -rf` like this.   What if I was to build with `SYSROOT_INC=/usr/local/include`..  this also makes me think is that `install` phase you want to be working with.

Reply:
I replied below to what you wrote just above.

As to your original question

> I think its we are building the sysroot from scratch we probably want remove the entire thing somewhere. (doesn't have to be here).

I made `make clean` rule below. I dislike the "sysroot" terminology, but I have no issue with wanting to make a rule to "start over"!